source/getting-started/sec-debugging.ptx

```diff
@@ -99,20 +72,176 @@
+      fill_op = dict(rgbcolor='blue', alpha=0.3)
+      options = dict(pathoptions=path_op, filloptions=fill_op, endarrow=False, startpoint=False)
+      G = [words.fibonacci_tile(i).plot(**options) for i in range(7)]
+      a = animate(G)
+      a.save(delay=150, 'LabbeFibonacciTiles.gif')
+    </input>
+  </sage>
+  <p>
+    In this example, <c>SyntaxError</c> occurs because the arguments in the <c>save()</c> method are in the wrong order. We can fix this by referring to the documentation and learning the correct argument positions. After running the following cell, the <c>save()</c> method saves the file to the cell. You can right-click on the file name and select the option to download the file locally. 
+  </p>
+  <sage>
+    <input>
+      # Originally created by Sébastien Labbé.
+      # Adapted from the original available at: https://wiki.sagemath.org/art licensed under GPLv2. For more information, visit: https://moinmo.in/GPL
+
+      path_op = dict(rgbcolor='red', thickness=1)
+      fill_op = dict(rgbcolor='blue', alpha=0.3)
+      options = dict(pathoptions=path_op, filloptions=fill_op, endarrow=False, startpoint=False)
+      G = [words.fibonacci_tile(i).plot(**options) for i in range(7)]
+      a = animate(G)
+      a.save('LabbeFibonacciTiles.gif', delay=150)
+    </input>
+  </sage>
+  <p>
+    Sometimes, Sage offers flexibility when using a method with dot notation or as a function. Here is another valid way of using <c>save()</c>.
+  </p>
+  <sage>
+    <input>
+      # Originally created by Sébastien Labbé.
+      # Adapted from the original available at: https://wiki.sagemath.org/art licensed under GPLv2. For more information, visit: https://moinmo.in/GPL
+
+      path_op = dict(rgbcolor='red', thickness=1)
+      fill_op = dict(rgbcolor='blue', alpha=0.3)
+      options = dict(pathoptions=path_op, filloptions=fill_op, endarrow=False, startpoint=False)
+      G = [words.fibonacci_tile(i).plot(**options) for i in range(7)]
+      a = animate(G)
+      save(a, 'LabbeFibonacciTiles.gif', delay=150)
+    </input>
+  </sage>
+  <p>
+    Finally, we will consider a logical error. If your task is to print the numbers from <m>1</m> to <m>10</m>, you may mistakenly write the following code:
+  </p>
+  <sage>
+    <input>
+      for i in range(10):
+          print(i)
+    </input>
+  </sage>
+  <p>
+    The output will be the numbers from <m>0</m> to <m>9</m>. To include <m>10</m>, we need to adjust the range because the start is inclusive and the stop is exclusive.
+  </p>
+  <sage>
+    <input>
+      for i in range(1, 11):
+          print(i)
+    </input>
+  </sage>
+  <p>
+    Remember Sage list comprehension behaves differently from <c>range()</c>. The range syntax for list comprehension includes the ending value.
+  </p>
+  <p>
+    For more information, read the CoCalc <url href="https://github.com/sagemathinc/cocalc/wiki/MathematicalSyntaxErrors">article about the top mathematical syntax errors in Sage</url>
+  </p>
```

Comment:
I am not sure about all this part. Maybe I will take a look once that is deployed and tell you later - not sure about its function here